mqtt-v5-broker/src/broker.rs

```diff
@@ -190,24 +223,58 @@ pub enum BrokerMessage {
     Disconnect(String, WillDisconnectLogic),
 }
 
-pub struct Broker {
+pub struct Broker<A> {
```

Comment:
Generic arguments can have a default value so maybe specifying our default plugin impl would be good here: 

`pub struct Broker<A = Noop>`